src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Common/XamarinProject.cs

```diff
@@ -33,7 +33,6 @@ public abstract class XamarinProject
 		public string GlobalPackagesFolder { get; set; } = FileSystemUtils.FindNugetGlobalPackageFolder ();
 		public IList<string> ExtraNuGetConfigSources { get; set; } = new List<string> ();
 
-		public virtual bool ShouldRestorePackageReferences => PackageReferences?.Count > 0;
```

Comment:
.NET 6+ projects always need to restore, so I removed this in `main` going forward.